shuup/default_reports/reports/customer_sales.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2017, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from operator import itemgetter
+
+from django.db.models import Avg, Count, Sum
+from django.utils.translation import ugettext_lazy as _
+
+from shuup.core.models import Contact
+from shuup.default_reports.forms import CustomerSalesReportForm
+from shuup.default_reports.mixins import OrderReportMixin
+from shuup.reports.report import ShuupReportBase
+
+
+class CustomerSalesReport(OrderReportMixin, ShuupReportBase):
+    identifier = "customer_sales_report"
+    title = _("Customer Sales")
+    form_class = CustomerSalesReportForm
+
+    filename_template = "customer-sales-report-%(time)s"
+    schema = [
+        {"key": "customer", "title": _("Customer")},
+        {"key": "order_count", "title": _("Orders")},
+        {"key": "average_sales", "title": _("Average Sales")},
+        {"key": "taxless_total", "title": _("Taxless Total")},
+        {"key": "taxful_total", "title": _("Taxful Total")},
+    ]
+
+    def get_objects(self):
+        orders = super(CustomerSalesReport, self).get_objects()
+
+        contacts = Contact.objects.values("pk", "name").order_by().filter(
```

Comment:
blank `order_by`. Also rather than the call to `sorted` below couldn't you use `order_by` after the `annotate`?